test/pet.test.js

```diff
@@ -241,9 +241,12 @@ describe('isAlive', () => {
 
 describe('haveBaby', () => {
 it ('Has a baby and pushes it into children[]', () => {
-  
-    petCat.haveBaby('kitten');
-    expect(petCat.children[0]).toBe(petCat.children['kitten']);
+    const child = new Pet ('Baby');
```

Comment:
This test will definitely pass, as it is, because you're creating a child object inside of it and passing it to the function. However, if, let's suppose, you had an interface with a button that on pressing would call the haveBaby() method. What would happen, according to the code you have at this point?